### PR TITLE
Share/pet card item

### PR DESCRIPTION
**Pull request by PiotrDrechsler**
## `PetCardItem` Component

The `PetCardItem` component displays detailed information about a pet, including its name, gender, and image. It also includes a button to view the pet's story. This component is designed with responsive styles and offers a visually structured card for each pet.

### Props

- **`name: string`**  
  The name of the pet. It will be prominently displayed on the card as the pet's identifier.

- **`gender: string`**  
  The gender of the pet, which can be either "female" or "male". Based on the value, an icon with a color specific to the gender is rendered to indicate the pet's gender.

- **`image: string`**  
  The URL for the pet's image. This image will be displayed within a rounded, framed area, providing a clear and appealing visual representation of the pet.

### Features

- **Responsive Card Design**  
  The component is styled with Tailwind CSS to ensure it adapts to different screen sizes. The card layout, padding, and font sizes adjust based on device width for an optimized display.

- **Semantic Accessibility**  
  The card includes `aria-labels` to provide accessible descriptions for screen readers. This includes descriptive text for the pet's name and gender, enhancing usability for visually impaired users.

- **Icon Integration**  
  The `IconPetGender` component is used to represent the pet's gender visually. The icon color changes based on the gender—pink for female pets and blue for male pets—using a utility class from a helper function (`cn`).

- **Button for Navigation**  
  A `Button` component labeled "Poznaj historię" ("Learn the story") navigates to the pet's story page. This button is styled for prominence and includes an accessible `aria-label` for screen readers, describing the action with the pet's name for better context.

### Example Usage

```jsx
<PetCardItem
  name="Lizi"
  gender="female"
  image="/images/fromLove/lizi/lizi01.jpg"
/>
```

### CODE REVIEW

File: src/components/Share/PetCardItem.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { z } from "zod";
+import Button from "./Button";
+import { IconPetGender } from "@/icons/IconPetGender";
+import { cn } from "@/utils/helpers";
+
+const petCardItemSchema = z.object({
+  name: z.string(),
+  gender: z.string(),
+  image: z.string().url(),
+});
```
Comment (NacomiTagiera) [inline src/components/Share/PetCardItem.tsx:11]: Gender can either be male or female, so you can modify the schema:
`const petCardItemSchema = z.object({
  name: z.string(),
  gender: z.enum(["female", "male"]),
  image: z.string().url(),
});`

File: src/components/Share/PetCardItem.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+export const PetCardItem: React.FC<PetCardItemProps> = ({
+  name,
+  gender,
+  image,
+}) => {
+  return (
+    <div
+      className="flex max-w-[242px] flex-col items-center rounded-[32px] border px-[16px] py-[24px] leading-[1.5] shadow-card-shadow tablet:max-w-[275px] tablet:py-[32px] desktop:px-[32px] desktop:py-[24px]"
+      aria-label={`${name} - ${gender === "female" ? "samiczka" : "samiec"}`}
+    >
```
Comment (NacomiTagiera) [inline src/components/Share/PetCardItem.tsx:24]: Consider using `<figure>` and  `<figcaption>` tags for better accessibility.

**Review by NacomiTagiera — CHANGES_REQUESTED** [review]

File: src/components/Share/PetCardItem.tsx (hunk 1 above)
Comment (PiotrDrechsler) [reply src/components/Share/PetCardItem.tsx]: fixed! i add PerProps in Types in utils

File: src/components/Share/PetCardItem.tsx (hunk 2 above)
Comment (PiotrDrechsler) [reply src/components/Share/PetCardItem.tsx:20]: fixed

**Review by NacomiTagiera — APPROVED** [review]

**Review by Delicadency — APPROVED** [review]
Nicely done component ❤️ Good work @PiotrDrechsler

**Pull request merged**